tensorflow/core/kernels/mkl/mkl_quantized_conv_ops_perchannel_test.cc

```diff
@@ -151,30 +155,35 @@ TEST_F(QuantizedConv2DPerchannelTest, Small) {
   AddInputFromArray<float>(TensorShape({1}), {image_max});
   AddInputFromArray<float>(TensorShape({2}), {filter_min, filter_min});
   AddInputFromArray<float>(TensorShape({2}), {filter_max, filter_max});
-  AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
-  AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
-  AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
-  AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
-  AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
-  AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+  if (!NativeFormatEnabled()) {
+    AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+    AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+    AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+    AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+    AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+    AddInputFromArray<uint8>(dummy_shape, dummy_tensor);
+  }
 
   // Run the op Kernel.
   TF_ASSERT_OK(RunOpKernel());
 
   // Get the output
   const Tensor& output = *GetOutput(0);
-  const Tensor& output_mkl_metadata = *GetOutput(3);
-
-  // Convert the output tensor in MKL to TF format.
-  ConvMklToTF conv_comp;
-  Tensor output_quantized;
-  conv_comp.ConvertMKL2TF<qint32>(DT_QINT32, output, output_mkl_metadata,
-                                  output_quantized);
-
   const float output_min = GetOutput(1)->flat<float>()(0);
   const float output_max = GetOutput(2)->flat<float>()(0);
-  Tensor output_float =
-      QuantizedTensorToFloat<qint32>(output_quantized, output_min, output_max);
+  Tensor output_quantized;
+  if (!NativeFormatEnabled()) {
+    const Tensor& output_mkl_metadata = *GetOutput(3);
+
+    // Convert the output tensor in MKL to TF format.
+    ConvMklToTF conv_comp;
+    conv_comp.ConvertMKL2TF<qint32>(DT_QINT32, output, output_mkl_metadata,
+                                    output_quantized);
+  }
+
+  Tensor output_float = QuantizedTensorToFloat<qint32>(
+      NativeFormatEnabled() ? output : output_quantized, output_min,
+      output_max);
```

Comment:
This is used often. Let's refactor this to a utility function.

Reply:
I added the refactored code to new file (mkl_kernel_util.h). I tried adding that code tensorflow/core/util/mkl_util.h but I had hard time trying to fix bazel circular dependency as this file depends on too many packages. So created this file where we can move common kernels-related code to.